### ISSUE TITLE
Question about focus-order-semantics logic

### ISSUE THREAD

**Issue by nolandanley** [dequelabs/axe-core#3863]
### Product

axe-core

### Question

I was looking into the [focus-order-semantics](https://dequeuniversity.com/rules/axe/4.5/focus-order-semantics) rule and found a minimal diff where one has a violation and the other doesn't, but I can't figure out why. Any insight would be helpful, both into the rule's logic and if one actually does have an accessibility issue the other doesn't. (Note: I'm using v4.5.2 but from looking at the release notes for v4.6.X I don't think this behavior has significantly changed.)

The first treegrid's row raises a violation, but re-implementing the same thing with HTML table elements doesn't raise a violation. These seem to have the same accessibility tree, and they seem to behave the same way when I interact with them with the various assistive tech I have access to, which makes me expect either they'd both raise a violation or neither would.

# Code
```html
<!-- 
  Copyright 2023 Google LLC.
  SPDX-License-Identifier: Apache-2.0
-->

<style>
  [role="treegrid"] {
    border: solid;
    width: fit-content;
  }
  
  [role="row"] {
    display: block;
    width: 100px;
  }
</style>

<div role="treegrid" tabindex="-1" aria-label="Navigation">
  <!-- violation -->
  <div role="row" aria-level="1" tabindex="0" aria-selected="true">
    <div role="gridcell"><a href="www.github.com">Github</a></div>
  </div>
</div>

<table role="treegrid" tabindex="-1" aria-label="Navigation">
  <!-- no violation -->
  <tr role="row" aria-level="1" tabindex="0" aria-selected="true">
    <td role="gridcell"><a href="www.github.com">Github</a></td>
  </tr>
</table>
```

# Accessibility Tree
Taken from Chrome's devtools's accessibility tree view.

* treegrid "Navigation" focusable:true multiselectable:false readonly:false required:false
    * row "Github" focusable:true selected:true
        * gridcell "Github" readonly:false required:false
            * link "Github" focusable:true
                * staticText "Github"
                    * InlineTextBox ""

**Comment by straker** [comment]
The reason the table does not throw a violation is because [the selector for `focus-order-semantics`](https://github.com/dequelabs/axe-core/blob/develop/lib/rules/focus-order-semantics.json#L3) does not look at `tr` elements. As for why that is I'm not sure. It's a pretty old rule and one that has been in experimental the entire time.

**Comment by WilcoFiers** [comment]
@straker @nolandanley It's pure heuristics. What it selects for are elements where people often put unnecessary tabindexes. The rule finds common problems, but it's not guaranteed to find all problems of its type. To some extend, that's kind of true for all rules in axe, and in other tools too. You can't catch everything, and at some point you have to apply some heuristics to avoid catching things you shouldn't.

The thing I'm learning from this example is that this rule should look at whether there's an explicit role. I don't think there's anything odd about having focusable rows, especially so in a `treegrid`.
